vavr/src/main/java/io/vavr/control/Try.java

```diff
@@ -972,6 +972,38 @@ default void orElseRun(Consumer<? super Throwable> action) {
         }
     }
 
+    /**
+     * Converts this {@code Try} to an {@link Validation}.
+     *
+     * @return A new {@code Validation}
+     */
+    default Validation<Throwable, T> toValidation() {
+        if (isFailure()) {
+            return Validation.invalid(getCause());
+        } else {
+            return Validation.valid(get());
+        }
+    }
+
+    /**
+     * Converts this {@code Try} to an {@link Validation}, converting the Throwable (if present)
+     * to another object using passed {@link Function}.
+     *
+     * <pre>
+     * <code>
+     * Validation<Failure, Integer> = Try.of(() -> 1/0).toValidation(t -> Failure.from(throwable.getMessage()));
+     * </code>
+     * </pre>
+     *
+     * @param throwableMapper  A transformation from throwable to desired invalid type of new {@code Validation}
+     * @return A new {@code Validation}
+     * @throws NullPointerException if the given {@code throwableMapper} is null.
+     */
+    default <U> Validation<U, T> toValidation(Function<Throwable, ? extends U> throwableMapper) {
+        Objects.requireNonNull(throwableMapper, "throwableMapper is null");
+        return toValidation().mapError(throwableMapper);
```

Comment:
I think it is better to switch implementations in order to save one intermediate Validation instance.

Here it is:

```java
if (isFailure()) {
    return Validation.invalid(throwableMapper.apply(getCause()));
} else {
    return Validation.valid(get());
}
```

Then the `toValidation()` method above looks like this:

```java
return toValidation(Function.identity());
```

I think the JIT magic will inline the identity() function, it will perform well.